Fixes and reworks.

vlams1 opened the pull request:
- Fixed pitch calculations.
- Fixed note count limit, now "used up" voices will be skipped, rather than stopping the entire process.
- Changed usage of LFOs in vital to remap curves to lower amount of LFOs needed for sequencing to one.
- Changed patch to use phase modulation rather than pitch modulation. (increasing the voice count to 6)
- Removed some LFO points from sequences to allow for a total of 45 notes per voice.
- Added commandline argument for midi file
- Updated readme

vlams1 committed:
6 voices working

Changed LFO generation to remap generation.

Set up base patch to work with phase modulation rather than pitching oscillators.

Removed "unnecessary" points from LFOs to allow for more notes

Improved LFO pitch calculation.

vlams1 committed:
Updated note lookup

Shortened note name to LFO pitch function

vlams1 committed:
Updated readme

added credits

updated limitations

vlams1 committed:
Fixed note limit issue

Global note limit has been replaced with a note limit per voice. This ups the amount of possible notes in a sequence to 270.

bluskript commented on src/index.ts:
```diff
@@ -1,49 +1,58 @@
 
-const midiData = readFileSync("./old-macdonald-had-a-farm.mid");
-const midi = new Midi(midiData);
+const midiFile = readFileSync(process.argv.slice(2)[0] || "./old-macdonald-had-a-farm.mid");
```
remove slice

bluskript commented on src/index.ts:
```diff
@@ -1,49 +1,58 @@
+const maxVoiceNotes = 45;
+
+for (let i = 0; i < voices*2; i++) {
+  data.settings.modulations[i].line_mapping = {} as Lfo;
```
this is a lot of repetitive accesses. might wanna store it in a variable

bluskript commented on src/lfoRepr.ts:
```diff
@@ -41,14 +20,15 @@ export function applyNote(
-  pitchLFO.points.push(end, pitch);
-  pitchLFO.points.push(end, 1.0);
+  console.log({pitch, name});
+  if (pitch == lastPitch)
```
strict equality pls `===`

bluskript commented on src/index.ts:
```diff
@@ -1,49 +1,58 @@
-  lfo.points.push(0, 1);
-  lfo.points.push(0, 0);
+
+const voices = 6;
```
we might wanna add these as command line options

vlams1 committed:
Removed slice

removed slice :)

bluskript commented on src/lfoRepr.ts:
```diff
@@ -2,37 +2,16 @@ import { Note } from "@tonejs/midi/dist/Note";
+  return 1 - pitch / 12 / 8;
 }
 
+let lastPitch;
```
this global variable should be un-globalified

vlams1 committed:
Restructuring and optimizations

created args constant.

Added console argument for voice count.

Fixed repetitive accessing.

Moved note repetition logging to index.

Bluskript commented:
LGTM
